src/experimental/ProbabilisticGraphicalModels/bayesian_network.jl

```diff
@@ -40,6 +44,76 @@ function BayesianNetwork{V}() where {V}
     )
 end
 
+"""
+    translate_BUGSGraph_to_BayesianNetwork(g::MetaGraph; init=Dict{Symbol,Any}())
+
+Translates a BUGSGraph (with node metadata stored in NodeInfo) into a BayesianNetwork.
+"""
+function translate_BUGSGraph_to_BayesianNetwork(g::MetaGraph)
+    # Retrieve variable labels (stored as VarNames) from g.
+    varnames = collect(labels(g))
+    n = length(varnames)
+    original_graph = g.graph
+
+    # Preallocate arrays/dictionaries.
+    names = Vector{VarName}(undef, n)
+    names_to_ids = Dict{VarName,Int}()
+    values = Dict{VarName,Any}()
+    distributions = Vector{Distribution}(undef, n)
```

Comment:
```suggestion
    distributions = Vector{Function}(undef, n)
```
and store node function. Also probably change the name `distributions`